File: UDPServer/Packet/UDPPacket.cpp
#include "UDPPacket.h"

void UDPPacket::read(char* buffer, int bufferSize, int& index)
{
	mType = Utility::read<PACKET_TYPE>(buffer, bufferSize, index);
}

void UDPPacket::write(char* buffer, int bufferSize, int& index)
{
	Utility::write<PACKET_TYPE>(buffer, bufferSize, index, mType);
}

File: UDPClient/Assets/Packet/UDPPacket.cs
using UnityEngine;
using static BinaryUtility;

public class UDPPacket
{
	protected PACKET_TPYE mType;
	public void setType(PACKET_TPYE type) { mType = type; }
	public PACKET_TPYE getType() { return mType; }
	public virtual void read(byte[] buffer, ref int index)
	{
		mType = (PACKET_TPYE)readByte(buffer, ref index, out _);
	}
	public virtual void write(byte[] buffer, ref int index)
	{
		writeByte(buffer, ref index, (byte)mType);
	}
	public virtual void execute() { }
}
